include/seqan3/range/detail/random_access_iterator.hpp

```diff
@@ -44,214 +44,263 @@
     using iterator_category = std::random_access_iterator_tag;
 
     /*!\name Constructors/Destructors
      * \{
-    */
-    // \brief Default constructor.
-    constexpr random_access_iterator() = default;
-
-    //!\brief Construct by host, default position pointer with 0.
-    explicit constexpr random_access_iterator(container_type & host) noexcept : host{&host} {}
-
-    //!\brief Construct by host and explicit position.
-    constexpr random_access_iterator(container_type & host, position_type const pos) noexcept : host{&host}, pos{pos} {}
-
+     */
+    //!\brief Default constructor.
+    constexpr random_access_iterator_base() = default;
     //!\brief Copy constructor.
-    constexpr random_access_iterator(random_access_iterator const &) = default;
-
+    constexpr random_access_iterator_base(random_access_iterator_base const &) = default;
     //!\brief Copy construction via assignment.
-    constexpr random_access_iterator & operator=(random_access_iterator const &) = default;
-
+    constexpr random_access_iterator_base & operator=(random_access_iterator_base const &) = default;
     //!\brief Move constructor.
-    constexpr random_access_iterator (random_access_iterator &&) = default;
-
+    constexpr random_access_iterator_base (random_access_iterator_base &&) = default;
     //!\brief Move assignment.
-    constexpr random_access_iterator & operator=(random_access_iterator &&) = default;
-
+    constexpr random_access_iterator_base & operator=(random_access_iterator_base &&) = default;
     //!\brief Use default deconstructor.
-    ~random_access_iterator() = default;
+    ~random_access_iterator_base() = default;
+
+    //!\brief Construct by host, default position pointer with 0.
+    explicit constexpr random_access_iterator_base(range_type & host) noexcept : host{&host} {}
+    //!\brief Construct by host and explicit position.
+    constexpr random_access_iterator_base(range_type & host, position_type const pos) noexcept :
+        host{&host}, pos{pos}
+    {}
 
     //!\brief Constructor for const version from non-const version.
-    template <typename t>
+    template <typename range_type2>
     //!\cond
-        requires std::is_same_v<std::remove_const_t<container_type>, t> &&
-                 std::is_same_v<container_type, std::add_const_t<t>>
+        requires std::is_const_v<range_type> && !std::is_const_v<range_type2> &&
+                 std::is_same_v<std::remove_const_t<range_type>, range_type2>
     //!\endcond
-    constexpr random_access_iterator(random_access_iterator<t> const & rhs) noexcept :
+    constexpr random_access_iterator_base(random_access_iterator_base<range_type2, derived_t_template> const & rhs) noexcept :
         host{rhs.host}, pos{rhs.pos}
     {}
     //!\}
 
     /*!\name Comparison operators
-     * \brief Compares only the absolute position of two iterators.
+     * \brief seqan3::detail::random_access_iterator_base operators are used unless specialised in derived type.
```

Comment:
I don't think this is a proper brief description. 
It is clear, that if you want to have special comparison operator you need to overload it.
This could be added as a general note in the class description.
Here I would say: `Compares two iterators for equality`.
What about the complexity? Is that documented in the class description?
What about exception safety?
What about return type?
What about parameter?
Don't we need to check this for these functions? Is doxygen not complaining, that param and stuff is not documented?
Please explain.

Reply:
I referred to the the base class here, because this piece of documentation will appear in the documentation of derived types, but I will update it to make it clearer.